homeassistant/components/qnap_qsw/config_flow.py

```diff
@@ -0,0 +1,65 @@
+"""Config flow for QNAP QSW."""
+from __future__ import annotations
+
+from typing import Any
+
+from aioqsw.exceptions import LoginError, QswError
+from aioqsw.localapi import ConnectionOptions, QnapQswApi
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME
+from homeassistant.data_entry_flow import AbortFlow, FlowResult
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import DOMAIN
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle config flow for a QNAP QSW device."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            url = user_input[CONF_URL]
+            username = user_input[CONF_USERNAME]
+            password = user_input[CONF_PASSWORD]
+
+            qsw = QnapQswApi(
+                aiohttp_client.async_get_clientsession(self.hass),
+                ConnectionOptions(url, username, password),
+            )
+
+            try:
+                system_board = await qsw.validate()
+            except LoginError:
+                errors["base"] = "invalid_auth"
```

Comment:
```suggestion
                errors[CONF_PASSWORD] = "invalid_auth"
```

Reply:
Changed in ccf3a61013a649c3bb9521ac2d1d1ee67e2ff281